packages/thirdweb/src/react/web/ui/prebuilt/NFTMedia/index.tsx

```diff
@@ -0,0 +1,97 @@
+import type { Chain } from "../../../../../chains/types.js";
+import { getContract } from "../../../../../contract/contract.js";
+import { punkImageSvg } from "../../../../../extensions/cryptopunks/__generated__/OnchainCryptoPunks/read/punkImageSvg.js";
+import { tokenURI } from "../../../../../extensions/erc721/__generated__/IERC721A/read/tokenURI.js";
+import { uri } from "../../../../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
+import type { Hex } from "../../../../../utils/encoding/hex.js";
+import { MediaRenderer } from "../../MediaRenderer/MediaRenderer.js";
+import type { MediaRendererProps } from "../../MediaRenderer/types.js";
+
+type NFTMediaProps = MediaRendererProps & {
+  // The NFT contract address
+  contractAddress: Hex;
+  // The chain which the NFT contract was deployed on
+  chain: Chain;
+  // The tokenId whose media you want to load
+  tokenId: bigint;
+
+  /**
+   * By standard format we look for the `image` field from the metadata (e.g: `const image = metadata.image`)
+   * However, some contracts do not follow the standard.
+   * We allow them to override that rule.
+   */
+  overrideMediaField?: string;
```

Comment:
I like it. It's low effort to add & maintain and it can be the decinding factor whether the component works or doesn't (I know at least one of a user of the marketplace template who has this onchain collection with the metadata that doesn't follow standard)

that said im open to any suggestion! cc @jnsdls @gregfromstl